src/App.js

```diff
@@ -1,23 +1,82 @@
-import logo from './logo.svg';
-import './App.css';
+import { useState } from "react";
+import "./App.css";
+import contactsJSON from "./contacts.json";
 
 function App() {
+  const [contacts, setContacts] = useState(contactsJSON.slice(0, 5));
+
+  function getRandomContact() {
+    const random = Math.floor(Math.random() * contactsJSON.length);
+
+    const randomContact = contactsJSON[random];
+
+    setContacts(contacts.concat(randomContact));
+  }
+
+  function sortByPopularity() {
+    const sortedContacts = contacts.sort((a, b) => b.popularity - a.popularity);
+    setContacts([...sortedContacts]);
+  }
+
+  function sortByName() {
+    const sortedByName = contacts.sort(function (a, b) {
+      if (a.name < b.name) {
+        return -1;
+      }
+      if (a.name > b.name) {
+        return 1;
+      }
+      return 0;
+    });
+    setContacts([...sortedByName]);
+  }
+
+  function deleteContact(contactId) {
+    const filteredContacts = contacts.filter((contact) => {
+      return contact._id !== contactId;
```

Comment:
Your delete function doesn't work properly, because there's no "_id". All of this data is coming from the "contacts.json" file, and, if you look at it, the field is called "id", without the underscore.

```suggestion
      return contact.id !== contactId;
```